coral/src/domain/topic/topic-types.ts

```diff
@@ -39,9 +39,30 @@ type TopicRequestApiResponse = ResolveIntersectionTypes<
   Paginated<TopicRequest[]>
 >;
 
+// KlawApiModel<"TopicOverview">
+// Represents the TopicOverview as defined in the backend.
+// "TopicOverview" is the type (and object) we're using in FE
+// we're redefining property types here to fit our need in app better
+// transformTopicOverviewResponse() is taking care of transforming
+// the properties and makes sure the types are matching between BE and FE
+type TopicOverview = ResolveIntersectionTypes<
+  Omit<
```

Comment:
I don't think we should cut off anything from the data we forward from `domain`. I know it is a contentious topic, but I've run in more problems with hiding available data than having too much available data. Like,  "I need this data but it's not there, let me create a contraption to get it", when it's actually there, but hidden by a transformer.

Exception would be `topicInfoList` where we want to surface the data out of its array :o

Additionally, `topicPromotionDetails` is actually used heavily in this PR ^^ https://github.com/aiven/klaw/pull/1353

Reply:
> Exception would be topicInfoList where we want to surface the data out of its array :o

If we want to avoid cutting out too much data, should we then have a property `topicInfo` (instead of *List) where the first entry can be found? That way the structure stays close to the API. We still would have to access `topicOverview.topicInfo`, but that's already better than `topicOverview.topicInfo[0]` :D